summerofcode/2023.md

```diff
@@ -141,3 +141,34 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
 - Difficulty: Medium
 - Upstream Issue (URL): https://github.com/strimzi/strimzi-kafka-operator/issues/8030
 
+### Curve
+
+#### Support metadata of Curve Block Storage stored in the database
+
+- Description: Currently the [metadata](https://github.com/opencurve/curve/blob/master/src/mds/nameserver2/namespace_storage.cpp) of current CurveBS is stored in [Etcd](https://github.com/opencurve/curve/tree/master/src/kvstorageclient). In order to apply to more scenarios, it is hoped that the metadata can add a sql database like mysql, MariaDB, PostgreSQL, etc. as one of the storage engines. Add a new sqlstorageclient for [metadata](https://github.com/opencurve/curve/blob/master/src/mds/nameserver2/namespace_storage.cpp).
+- Expected outcome: It is hoped that the user can choose whether the metadata is stored in the kv engine or the sql engine through the configuration file.
+- Recommended Skills: C++, SQL
+- Mentor(s): Xiaocui Li @ilixiaocui
+- Expected project size: 175h
+- Difficulty: Medium
+- Upstream Issue (URL): https://github.com/opencurve/curve/issues/2224
+
+#### Add compression feature for Curve snapshot
+
+- Description: Curve snapshots are currently uploaded to s3, and no compression is used. Using compression can greatly reduce the space occupied by snapshots, thereby saving storage costs.
+- Expected outcome: 1） Detailed Design Documentation 2） Realize curve snapshot compression upload and download decompression 3） Add unit tests and integration test cases 4） Merge your PR into the opencurve repo
+- Recommended Skills: C++
+- Mentor(s): Chaojie Xu @xu-chaojie
+- Expected project size: 175 hours
+- Difficulty: Medium
+- Upstream Issue (URL): https://github.com/opencurve/curve/issues/2223
+
+#### Add Tracing feature for Curve
+
+- Description: At present, Curve has logging and metrics, both of which can be used to analyze performance as well as locate problems. While they improve the observability of the system, the granularity is coarse and does not allow for precise analysis of how long requests take at each stage. Tracing is a powerful tool that can concatenate invocation relationships between services and log invocation time in the request dimension, preserving essential information and concatenating dispersed log events to help us better understand system behavior, assist in debugging and troubleshooting performance issues.
+- Expected outcome: Design the solution and implement it, introduce it into CurveBS, and analyze the latency of IO requests. The implementation needs to be well scalable and can be applied to other modules.
+- Recommended Skills: C++, OpenTracing
+- Mentor(s): Hanqing Wu @wu-hanqing 
```

Comment:
👍 like the previous mentor

Reply:
both Hanqing Wu and Chaojie Xu are [commiters](https://github.com/opencurve/community/blob/master/teams/Curve/team.json) in curve community.